components/sidebar/sidebar_service.cc

```diff
@@ -327,24 +315,26 @@ void SidebarService::RemoveObserver(Observer* observer) {
 }
 
 std::vector<SidebarItem> SidebarService::GetHiddenDefaultSidebarItems() const {
+  const auto added_default_items = GetCurrentlyPresentBuiltInTypes();
   auto default_items = GetDefaultSidebarItems();
-  const auto added_default_items = GetDefaultSidebarItemsFromCurrentItems();
-  for (const auto& added_item : added_default_items) {
-    auto iter = base::ranges::find_if(
-        default_items, [&added_item](const auto& default_item) {
-          return default_item.built_in_item_type ==
-                 added_item.built_in_item_type;
-        });
-    default_items.erase(iter);
-  }
```

Comment:
Q. So, Is this the root cause of the ASAN failure? I'm trying to wrap my head around this problem. I know erase-remove-if idiom, but not quite sure why the old code was the problem. Isn't it okay to call even `vector::erase(default_items.end())` ?

Reply:
The old cold was erasing `end()` which is invalid. This version of the code returns an empty range if no corresponding element was found, which results in a nullopt.

Reply:
Ah, it seems to be UB

Reply:
Yep. This one is pretty bad, but luckily ASAN caught it. I think this PR should be merged before the other PR dealing with the preferences crash.